File: football_field_manager/src/main/java/com/example/football_field_manager/dto/response/TimeSlotResponse.java
package com.example.football_field_manager.dto.response;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.time.LocalTime;
import java.util.Date;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
@JsonInclude(JsonInclude.Include.NON_NULL)
public class TimeSlotResponse {
    Long id;
    LocalTime startTime;
    LocalTime endTime;
    double price;
    Date onCreate;
    Date onUpdate;
}

File: football_field_manager_demo/lib/data/models/response/time_slot_response.dart
class TimeSlotResponse {
  int? id;
  String? startTime;
  String? endTime;
  double? price;
  String? onCreate;
  String? onUpdate;

  TimeSlotResponse({this.id, this.startTime, this.endTime, this.price, this.onCreate, this.onUpdate});

  TimeSlotResponse.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    startTime = json['startTime'];
    endTime = json['endTime'];
    price = json['price'];
    onCreate = json['onCreate'];
    onUpdate = json['onUpdate'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['startTime'] = this.startTime;
    data['endTime'] = this.endTime;
    data['price'] = this.price;
    data['onCreate'] = this.onCreate;
    data['onUpdate'] = this.onUpdate;
    return data;
  }
}
